src/Api/JsonApi/RequestUri.cs

```diff
@@ -0,0 +1,60 @@
+using System.Text;
+
+namespace Defra.PhaImportNotifications.Api.JsonApi;
+
+public record RequestUri(
+    string Path,
+    FilterExpression? Filter = null,
+    FieldExpression[]? Fields = null,
+    int? PageSize = null
+)
+{
+    public override string ToString()
+    {
+        var result = new StringBuilder();
```

Comment:
Can you use QueryStringBuilder?

Reply:
I tried that originally but backed away from it as something from JSON.API prevented it from being used. I'll revisit and let you know, thanks

Reply:
The reason I backed away from it was because it mandates a `key` and `value` to be specified where we have a field expression for JSON.API that produces the full key/value pair as a single string, which I was just appending ie. `fields[type]=field1,field2` etc.

I've just looked at using QueryBuilder again and it's not fit for purpose in relation to how it escapes. It's using UrlEncoder.Default and comparative escaping between that and what we are using is as follows:

`Uri.EscapeDataString(Filter.ToString())` => `equals%28name%2C%27Some%20Name%27%29`

`UrlEncoder.Default.Encode(Filter.ToString())` => `equals(name,%27Some%20Name%27)`

I want us to be quite aggressive with our URL escaping to BTMS due to its querying syntax. However, it looks like it's WireMock that is struggling to interpret a URL parameter containing a comma as it treats it as a separator. If we are aggressive with escaping and escape the comma, WireMock works and the request to BTMS works.

I'll do a bit more digging in WireMock to see if there is something we can do but as it stands, we can't use QueryBuilder immediately.

Reply:
Looks like it can resolved is comma is removed from the type of separators WireMock is using. Via:

```
new WireMockServerSettings
{
    QueryParameterMultipleValueSupport = QueryParameterMultipleValueSupport.NoComma,
}
```

This means that we never expect multi value query values such as `key=value1,value2` in a query string.